### PR TITLE
Anton/feature/search

### PR DESCRIPTION
**Pull request by sa1am8**
search route + pagination function

### CODE REVIEW

File: tests/test_routes/test_job.py [hunk 1]
```diff
@@ -133,3 +135,43 @@ def test_create_job(
...
+def test_search_job(
+    client: TestClient,
+    db: Session,
+):
+    fill_test_data(db)
+    create_professions(db)
+    create_jobs(db)
+
+    response = client.get("api/job/search")
+    assert response.status_code == 200
```
Comment (BohdanSizomin) [inline tests/test_routes/test_job.py:149]: status code ... from fastapi

**Review by BohdanSizomin — CHANGES_REQUESTED** [review]

**Review by BohdanSizomin — APPROVED** [review]

File: tests/test_routes/test_job.py [hunk 2]
```diff
@@ -4,13 +4,15 @@
 from fastapi import status
 from fastapi.testclient import TestClient
 from sqlalchemy.orm import Session
+from sqlalchemy import select
```
Comment (kipod) [inline tests/test_routes/test_job.py:7]:
```python
import sqlalchemy as sa
from sqlalchemy import orm
```

File: tests/test_routes/test_job.py [hunk 3]
```diff
@@ -139,3 +141,43 @@ def test_create_job(
...
+
+    response = client.get("api/job/search")
+    assert response.status_code == status.HTTP_200_OK
+    response_jobs_list = s.ListJob.parse_obj(response.json())
+    assert len(response_jobs_list.jobs) > 0
+
+    job: m.Job = db.scalar(select(m.Job))
+    assert job
+
+    response = client.get("api/job/search", params={"city": f"{job.city}"})
```
Comment (kipod) [inline tests/test_routes/test_job.py:162]: :) 
```python
response = client.get("/api/job/search", params=dict(city=job.city))
```

**Review by kipod — COMMENTED** [review]
Looks good

**Pull request merged**